core/src/conductor_api.rs

```diff
@@ -0,0 +1,59 @@
+use holochain_core_types::error::HolochainError;
+use jsonrpc_core::IoHandler;
+use jsonrpc_lite::JsonRpc;
+use snowflake::ProcessUniqueId;
+use std::{
+    fmt,
+    sync::{Arc, RwLock},
+};
+
+#[derive(Clone)]
+pub struct ConductorApi(Arc<RwLock<IoHandler>>);
+
+impl ConductorApi {
+    pub fn new(conductor_api: Arc<RwLock<IoHandler>>) -> ConductorApi {
+        ConductorApi(conductor_api)
+    }
+
+    pub fn sign(&self, payload: String) -> Result<String, HolochainError> {
+        let handler = self.0.write().unwrap();
+        let request = format!(
+            r#"{{"jsonrpc": "2.0", "method": "agent/sign", "params": {{"payload": "{}"}}, "id": "{}"}}"#,
+            payload, ProcessUniqueId::new(),
+        );
+
+        let response = handler
+            .handle_request_sync(&request)
+            .ok_or("Conductor sign call failed".to_string())?;
+
+        let response = JsonRpc::parse(&response)?;
+
+        match response {
+            JsonRpc::Success(_) => Ok(String::from(
+                response.get_result().unwrap()["signature"]
+                    .as_str()
+                    .unwrap(),
```

Comment:
I think we should handle the error here instead of unwrapping it here. Since the method already returns a result. Also I think accessing slices like this might invoke a panic as well